### PR TITLE
fix some memory leaks

### PR DESCRIPTION
**Pull request by alvarofe**
i would create an issue for 0.10.3 to reduce memory leaks because we are leaking a lot.

### CODE REVIEW

File: libr/anal/var.c [hunk 1]
```diff
@@ -370,7 +370,7 @@ R_API RList *r_anal_var_list(RAnal *a, RAnalFunction *fcn, int kind) {
 	char *varlist;
 	RList *list = r_list_new ();
 	if (!a || !fcn)
-		return list;
+		return NULL;
```
Comment (radare) [inline libr/anal/var.c]: missing r_list_free.. or well. we should construct `list = r_list_new` after this check. not before

File: libr/io/io.c [hunk 2]
```diff
@@ -992,13 +992,13 @@ static ut8 *r_io_desc_read(RIO *io, RIODesc *desc, ut64 *out_sz) {
 	off = io->off;
 
 	if (*out_sz == UT64_MAX) {
-		return buf;
+		return NULL;
```
Comment (radare) [inline libr/io/io.c]: buf is already null :P

Comment (alvarofe) [inline libr/io/io.c]: just to make it explicit in case in the future buf is set before or something

File: libr/anal/var.c (hunk 1 above)
Comment (alvarofe) [inline libr/anal/var.c]: yes, you are rigth i overlooked it

**Comment by radare** [comment]
Thanks!

**Pull request closed without merging**